src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractTestVector.java

```diff
@@ -0,0 +1,106 @@
+package com.sap.sgs.phosphor.fosstars.model.qa;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.sap.sgs.phosphor.fosstars.model.Interval;
+import com.sap.sgs.phosphor.fosstars.model.Label;
+import java.util.Objects;
+
+/**
+ * A base class for test vectors.
+ */
+public abstract class AbstractTestVector implements TestVector {
+
+  /**
+   * An interval for an expected score.
+   */
+  final Interval expectedScore;
+
+  /**
+   * If it's set to true, then a not-applicable score value is expected.
+   */
+  final boolean expectedNotApplicableScore;
+
+  /**
+   * An expected label.
+   */
+  final Label expectedLabel;
+
+  /**
+   * An alias of the test vector.
+   */
+  final String alias;
+
+  /**
+   * Initializes a new {@link StandardTestVector}.
+   *
+   * @param expectedScore An interval for an expected score.
+   * @param expectedLabel An expected label (can be null).
+   * @param alias A alias of the test vector.
+   * @param expectedNotApplicableScore
+   *        If it's set to true, then a not-applicable score value is expected.
+   */
+  AbstractTestVector(Interval expectedScore, Label expectedLabel, String alias,
+      boolean expectedNotApplicableScore) {
+
+    Objects.requireNonNull(alias, "Hey! alias can't be null!");
+
+    if (expectedScore == null && !expectedNotApplicableScore) {
```

Comment:
expectedNotApplicableScore could someone set this as `null`?

Reply:
Yes, `expectedScore` and `expectedLabel` are allowed to be null. `expectedLabel` may be null when a test vector is for a score (score don't produce labels). `expectedScore` may be null if an N/A value is expected (`expectedNotApplicableScore` should be true).

`expectedNotApplicableScore` can't be null since it's `boolean`.

Reply:
As a little improvement, we can use `Optional` to avoid null-checks.